### PR TITLE
engines: universal threshold post-step + analyzer aggregation helpers

### PR DESCRIPTION
**Pull request by deanlorenz**
## Summary

- Engine post-step applies a pure threshold formula to `RequiredCapacity` / `SpareCapacity` at model level and every `RoleCapacity` entry, replacing the saturation-only override-resolution loop. `TotalAnticipatedSupply` is read as-is — zero is a literal value, not a sentinel; no fallback walk over `VariantCapacities`.
- New `internal/engines/aggregation/` package with pure helpers (`SumTotalSupply`, `SumTotalAnticipatedSupply`, `SumTotalDemand`, `AggregateByRole`) over `[]VariantCapacity`. Analyzers use these to populate per-scope `Total*` fields and enforce the linearity invariant the optimizer's per-variant scaling math depends on.
- Saturation V2 simplified: drops in-analyzer RC/SC computation (engine post-step is now the sole writer); calls helpers for per-scope aggregation. Phase 4 redundancy removed.
- Per-analyzer threshold overrides resolved per analyzer (`AnalyzerScoreConfig.ScaleUpThreshold` / `ScaleDownBoundary` over the model-level globals) and applied uniformly at every scope. No per-role overrides.
- Dev-guide rewritten to document the architecture: per-variant `VariantCapacity` is canonical; analyzer publishes per-variant primitives + per-scope `Total*`; engine post-step computes RC/SC; `SchedulerQueue` is shared input across analyzers, but demand extraction is per-analyzer.

## Context

This PR is **stacked on #1225** (\`engines/saturation: multi-analyzer registry\`). The 4 threshold commits sit on top of #1225's 3 registration commits. When #1225 merges, this branch will be rebased onto main and the diff will reduce to the 4 threshold commits.

Implements Item 2 of the design discussion in \`planning/PR1113-review.md\`. Items 1 (combine deletion / per-analyzer slice → optimizers) and 3 (analyzer registration) are tracked separately.

## Test plan

- [x] \`make build\` — clean
- [x] \`golangci-lint run\` — 0 issues
- [x] \`make test\` — all packages pass
- [x] \`go test -race ./internal/engines/saturation/...\` — clean
- [x] \`internal/engines/aggregation/aggregation_test.go\` — full coverage of helper edge cases (empty input, single/multi variant, mixed roles, empty-role canonicalization, zero replicas, zero PRC)
- [x] \`internal/engines/saturation/engine_v2_threshold_test.go\` — pure-formula specs at model + per-role (12 specs covering scale-up/down, hysteresis band, exact-boundary clamps, anticipated-vs-steady asymmetry, literal-zero \`TotalAnticipatedSupply\`, non-positive thresholds, idempotency, nil-result, P/D disaggregation)
- [x] DCO sign-off on all 4 commits

## Behavior change

\`AnalyzerResult.RequiredCapacity\` and \`SpareCapacity\` are now **written exclusively by the engine post-step**. Analyzer-written values are discarded. Saturation V2's in-analyzer RC/SC computation has been removed; the engine's universal formula produces the same values for the same inputs. Any analyzer extending this path must populate per-scope \`Total*\` fields (use the shared helpers) and leave RC/SC zero.

### CODE REVIEW

File: internal/engines/analyzers/saturation_v2/analyzer.go [hunk 1]
```diff
@@ -417,15 +403,19 @@ func (a *SaturationAnalyzer) aggregateByVariant(
 }
 
 // aggregateByRole groups variant capacities by P/D role and computes per-role
-// scaling signals. Returns nil when no disaggregation is active (all variants
-// are role "both" or empty). The queueDemandByRole map adds scheduler queue
-// demand attributed to each role (nil when there's no queue demand).
+// aggregateByRole groups variant capacities by role and returns per-role
```
Comment (ev-shindin) [inline internal/engines/analyzers/saturation_v2/analyzer.go:405]: Duplicate comment `//AggregateByRole` please combine lines 405 and 406

**Review by ev-shindin — COMMENTED** [review]
This branch was cut before #1225's last two commits (`dd9834d`, `6339e49`), so the current diff shows `RegisterAnalyzer` back to `panic` and `runRegisteredAnalyzer` logging at `Error`. Nothing to do now — just, when you rebase onto the merged #1225.

Architecturally LGTM — the only code change needed is the one-line doc fix;.

**Review by ev-shindin — APPROVED** [review]
Thanks @deanlorenz !
